src/arrayMethodSort.js

```diff
@@ -2,11 +2,40 @@
 
 /**
  * Implement method Sort
- */
+//  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    let value1 = '';
+    let value2 = '';
+
+    if (!compareFunction) {
```

Comment:
to avoid this condition you can add a default parameter for compareFunction to compare as strings
```(a, b) => String(a) > String(b)```

```suggestion
```

Reply:
It's very clever. Done!